python/tvm/autotvm/task/space.py

```diff
@@ -119,21 +125,26 @@ def __init__(self, var, name=None):
         super(VirtualAxis, self).__init__()
         self.num_output = 1
 
-        if name is None:
-            name = 'axis_%d' % VirtualAxis.name_ct
-            VirtualAxis.name_ct += 1
+        self.name = None
+        if name is not None:
+            self.name = name
 
-        self.name = name
         if isinstance(var, (int, _long)):
             self.length = var
+            if name is None:
+                self.name = 'axis_%d' % VirtualAxis.name_ct
+                VirtualAxis.name_ct += 1
         elif isinstance(var, schedule.IterVar):
-            self.name = var.var.name
+            if self.name is None:
+                self.name = var.var.name
             if var.dom is None:
                 self.length = -1
             else:
                 self.length = get_const_int(var.dom.extent)
         elif isinstance(var, VirtualAxis):
             self.length = var.length
+            if self.name is None:
+                self.name = var.name
```

Comment:
Actually I'm a bit confused by the original semantic. It meant to overwrite the name with var's name if var is an IterVar, but not to overwrite if var is a VirtualAxis which already has a name. Plus, when name=None and var is an IterVar, name_ct increments by 1 but it's not used in naming. Is this what it is supposed to be?

Reply:
In my opinoin, the `name_ct` doesn't really matter because it just makes sure every name is unique, but I am not quite sure about the correct behavior to determine the name either.

@merrymercy could you help clarify?